### DISCUSSION TITLE
How to generate non-HTML file (SSG)

### DISCUSSION THREAD

**Discussion by iacore** [vikejs/vike#1521] [Discussion] [answered]
I want to create a file with the content `abc`. What can I do? `+Page` does not work here.

I kind of got it to work with

```
// +onRenderHtml.js
// Environment: server

export { onRenderHtml }

import type { OnRenderHtmlAsync } from 'vike/types'
import { dangerouslySkipEscape } from 'vike/server'

const onRenderHtml: OnRenderHtmlAsync = async (pageContext): ReturnType<OnRenderHtmlAsync> => {
  const documentHtml = dangerouslySkipEscape(`awawa`)

  return {
    documentHtml,
    pageContext: {},
  }
}
```

but the response has dev script/snippet mixed in. response:

```
<head><link rel="stylesheet" type="text/css" href="/pages/style.scss?direct"></head>
awawa
<script id="vike_pageContext" type="application/json">{"abortReason":"!undefined","_urlRewrite":null,"_urlRedirect":"!undefined","abortStatusCode":"!undefined","_abortCall":"!undefined","_pageContextInitIsPassedToClient":"!undefined","_pageId":"/pages/feed","data":"!undefined"}</script><script type="module" async>
import("/@vite/client");
import("/@fs/home/user/meta/www.identity/node_modules/.pnpm/vike@0.4.162_vite@5.1.3/node_modules/vike/dist/esm/client/client-routing-runtime/entry.js");
</script>
```

I want a way to turn it off.

Also, if the route folder is called `x.json`, then the route won't be loaded correctly by vite. 

And, also, the route string is not respected. There is no final slash, so the page should be put at `/client/feed.json`

![image](https://github.com/vikejs/vike/assets/74560659/8c38ba31-4696-4f87-a451-071e13d639be)

**Comment by brillout** [comment]
Why do you want to do that?

**Reply by iacore** [reply]
RSS/Atom/JSON Feed. I have a static website made with vike, and I want to add one of those feed types. I can write a script that generates it in `public/`, but I don't know how to integrate it with vike.

**Reply by brillout** [reply]
How about using a post-build script?

**Reply by iacore** [reply]
How do I run the post-build script in vite context? I need `import.meta.glob`.

**Reply by brillout** [reply]
I'm not sure if/how you can do that. A new CLI command `$ vike run ./some-script.ts` could be nice.

**Comment by lukajincharadze** [comment]
Mixed Content in Response:

The mixed content in your response is likely due to the default behavior of Vite to inject client-side scripts and snippets into the HTML response.
To disable this behavior and have full control over the generated HTML response, you may need to adjust your Vite configuration or the way you handle server-side rendering.
Review the Vite documentation and configuration options to see if there's a setting to disable or customize the injected scripts.
Route Folder Name Issue:

If naming your route folder x.json causes issues with routing and loading, it may be due to conflicts with file extensions or unexpected behavior in the routing system.
Try renaming the route folder to something without a file extension, such as x_route or x_folder, to see if the issue persists.
Respecting Route String:

If the route string is not being respected and doesn't include the final slash as expected, you may need to adjust your routing configuration or server logic.
Ensure that your server or routing system properly handles route strings and generates the correct URLs for your pages.
Double-check your routing configuration to see if there are any inconsistencies or missing settings.

**Reply by iacore** [reply]
I simply used the default routing setting provided by Vike. There is no server or routing configuration to be adjusted by me.

A solution to my issue was provided in https://github.com/vikejs/vike/issues/1525

**Comment by iacore** [comment, marked as answer]
As discussed in https://github.com/vikejs/vike/issues/1525, this use case is not yet supported by Vike.

## Workaround

Create a custom script that generate files run it with `vite-node generator-script.ts`.
